### PR TITLE
fix: Add button vars to pills to overwrite styles

### PR DESCRIPTION
**Pull request by mannycarrera4**
## Summary
BaseButton is bleeding through styles because the Pill doesn't define css variables for hover, causing the visual error on hover. Adding pointer events none prevents this from happening but doesn't fix the underlying issue.

Fixes: https://github.com/Workday/canvas-kit/issues/2428

## Release Category
Components

---

### CODE REVIEW

File: modules/preview-react/pill/lib/Pill.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import {CSSObject} from '@emotion/react';
 
-import {BaseButton} from '@workday/canvas-kit-react/button';
+import {BaseButton, buttonVars} from '@workday/canvas-kit-react/button';
```
Comment (mannycarrera4) [inline modules/preview-react/pill/lib/Pill.tsx:4]:
```suggestion
import {BaseButton} from '@workday/canvas-kit-react/button';
```

File: modules/preview-react/pill/lib/Pill.tsx [hunk 2]
```diff
@@ -19,6 +19,7 @@ import {PillIconButton} from './PillIconButton';
 import {PillCount} from './PillCount';
 import {PillAvatar} from './PillAvatar';
 import {PillLabel} from './PillLabel';
+import {createStyles} from '@workday/canvas-kit-styling';
```
Comment (mannycarrera4) [inline modules/preview-react/pill/lib/Pill.tsx:22]:
```suggestion
```

**Review by alanbsmith — DISMISSED** [review]
Thanks, Manny!

File: modules/styling/lib/cs.ts [hunk 3]
```diff
@@ -108,6 +107,18 @@ export function cssVar(input: string, fallback?: string) {
     : `var(${input})`;
 }
 
+/**
+ *
+ * @param key
+ * @returns
+ */
```
Comment (NicholasBoll) [inline modules/styling/lib/cs.ts:114]: I found where we implemented this before. I'm not sure how it got removed: https://github.com/Workday/canvas-kit/blob/b775c5737e0019395b32fa18dbd01df9008fca91/modules/react/common/lib/cs.ts#L84-L94

```suggestion
/**
 * Util function to fix an issue with Emotion by
 * appending `EmotionIssue#3066` to end of css variable
 * See issue: [#3066](https://github.com/emotion-js/emotion/issues/3066)
 */
```

**Review by jamesfan — APPROVED** [review]

**Pull request merged**